src/coreclr/jit/hwintrinsic.cpp

```diff
@@ -1657,14 +1657,14 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
             // HWInstrinsic requires a mask for op2
             if (!varTypeIsMask(op2))
             {
-                retNode->AsHWIntrinsic()->Op(2) = gtNewSimdCvtVectorToMaskNode(retType, op2, simdBaseJitType, simdSize);
+                retNode->AsHWIntrinsic()->Op(2) = gtNewSimdCvtVectorToMaskNode(TYP_MASK, op2, simdBaseJitType, simdSize);
```

Comment:
probably add an assert in `gtNewSimdCvtVectorToMaskNode()` that `type == TYP_MASK`?

Reply:
That assert already exists and is what caught the failure.